Source/Animator.swift

```diff
@@ -34,43 +38,46 @@ class Animator: NSObject {
   /// :param: data The raw GIF image data.
   /// :param: delegate An `Animatable` delegate.
   required init(data: NSData, delegate: Animatable) {
-    let imageSource = CGImageSourceCreateWithData(data, nil)
+    imageSource = CGImageSourceCreateWithData(data, nil)
     self.delegate = delegate
     super.init()
     attachDisplayLink()
-    curry(prepareFrames) <^> imageSource <*> delegate.frame.size
+    prepareFrames()
     pauseAnimation()
   }
 
+  deinit {
```

Comment:
I'm currently debugging this ... it's not fully working ... I found this out after making the PR ... sorry ... we can close this if that makes sense